### PR TITLE
HIVE-25022: Metric about incomplete compactions

### PR DESCRIPTION
**Pull request by klcopp**
### What changes were proposed in this pull request?
See https://issues.apache.org/jira/browse/HIVE-25022

### How was this patch tested?
Unit tests

### CODE REVIEW

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/metrics/AcidMetricService.java [hunk 1]
```diff
@@ -108,19 +112,56 @@ private void updateDBMetrics() throws MetaException {
...
       lastElements.compute(key, (k, old) -> (old == null) ? element : (element.getId() > old.getId() ? element : old));
       if (TxnStore.INITIATED_RESPONSE.equals(element.getState()) && oldestEnqueueTime > element.getEnqueueTime()) {
         oldestEnqueueTime = element.getEnqueueTime();
       }
+
+      // Count incomplete compactions
+      CompactionType type = element.getType();
+      lastUnsuccessfulMajor.compute(key, (k, old) -> {
+        // Add newest unsuccessful compaction to the map
+        if (wasUnsuccessful(element) && type == MAJOR) {
```
Comment (deniskuzZ) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/metrics/AcidMetricService.java:137]: Could it be simplified ?
```
if (type == MAJOR){
    if (wasUnsuccessful(element)) return element.getId();
    if (wasSuccessful(element)) return null;
}
```

**Review by deniskuzZ — COMMENTED** [review]
LGTM, however, I don't see real benefit from this metric. 
```
major succeeded
major failed
minor failed
minor succeeded
```
This would be reported as incomplete compaction. What action do you expect from the end-users in this case?

**Comment by klcopp** [comment]
> LGTM, however, I don't see real benefit from this metric.
> 
> ```
> major succeeded
> major failed
> minor failed
> minor succeeded
> ```
> 
> This would be reported as incomplete compaction. What action do you expect from the end-users in this case?

In this case end users should re-run major compaction. Major compaction should have run (at 2. major failed) but hasn't since that failure.

**Comment by deniskuzZ** [comment]
> > LGTM, however, I don't see real benefit from this metric.
> > ```
> > major succeeded
> > major failed
> > minor failed
> > minor succeeded
> > ```
> > 
> > 
> > This would be reported as incomplete compaction. What action do you expect from the end-users in this case?
> 
> In this case end users should re-run major compaction. Major compaction should have run (at 2. major failed) but hasn't since that failure.

How would they know tables/partitions to re-run major compaction?

**Pull request closed without merging**